subsys/net/lib/icalendar_parser/src/icalendar_parser.c

```diff
@@ -0,0 +1,427 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdio.h>
+#include <strings.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <toolchain/common.h>
+#include <net/icalendar_parser.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(icalendar_parser, CONFIG_ICAL_PARSER_LOG_LEVEL);
+
+static size_t unfold_contentline(const char *buf, char *prop_buf)
```

Comment:
It seems to me you're just trimming EOL sequences from the buffer here.
Does this function really need a separate output buffer, could it not modify the input buffer instead?